src/etc/inc/services.inc

```diff
@@ -1190,6 +1201,10 @@ EOD;
 				}
 
 				$dhcpdconf .= "}\n";
+
+				// subclass for DHCP limiting
+				$dhcpdconf .= "subclass \"s_{$dhcpif}\" 1:{$sm['mac']};\n"; // assuming ALL addresses are ethernet hardware type ("1:" prefix)
```

Comment:
What if the MAC is empty? Entries can be specified based on MAC or Client ID. Either one could be empty.

Reply:
MAC or Client ID, or both are added to config now